dynamiqs/mesolve/rouchon.py

```diff
@@ -24,43 +27,43 @@ class MERouchon1(MERouchon):
-            # self.M0, self.M0rev: (b_H, 1, n, n)
-            self.M0 = cache(lambda Hnh: self.I - 1j * self.dt * Hnh)
-            self.M0rev = cache(lambda Hnh: self.I + 1j * self.dt * Hnh)
-
-            # define M1s and M1srev
-            # self.M1s: (1, len(L), n, n)
-            self.M1s = sqrt(self.dt) * self.L
-            self.M1srev = self.M1s
-        else:
-            # compute the inverse square root renormalization matrix
-            # (for time-dependent Hamiltonians, exclude the Hamiltonian from
-            # normalization)
-            # Hnh, M0, M0rev, self.S, self.Srev: (b_H, 1, n, n)
-            if isinstance(self.H, ConstantTDTensor):
-                Hnh_const = self.Hnh(0.0)
-            elif isinstance(self.H, CallableTDTensor):
-                Hnh_const = -0.5j * self.sum_LdagL
-
-            M0 = self.I - 1j * self.dt * Hnh_const
-            M0rev = self.I + 1j * self.dt * Hnh_const
-            self.S = inv_sqrtm(M0.mH @ M0 + self.dt * self.sum_LdagL)
-            self.Srev = inv_sqrtm(M0rev.mH @ M0rev - self.dt * self.sum_LdagL)
-
-            # define cached operators
-            # self.M0, self.M0rev: (b_H, 1, n, n)
-            self.M0 = cache(lambda Hnh: self.S - 1j * self.dt * Hnh @ self.S)
-            self.M0rev = cache(lambda Hnh: self.Srev + 1j * self.dt * Hnh @ self.Srev)
-
-            # define M1s and M1srev
-            # self.M1s, self.M1srev: (1, len(L), n, n)
-            self.M1s = sqrt(self.dt) * self.L @ self.S
-            self.M1srev = sqrt(self.dt) * self.L @ self.Srev
-
-        self.M0dag = cache(lambda M0: M0.mH)  # (b_H, 1, n, n)
-        self.M1sdag = self.M1s.mH  # (1, len(L), n, n)
+        # forward time operators
+        # M0: (b_H, 1, n, n)
+        # M1s: (1, len(L), n, n)
+        self.M0 = cache(lambda Hnh: self.I - 1j * self.dt * Hnh)
+        self.M1s = sqrt(self.dt) * self.L
+
+        # reverse time operators
+        # M0rev: (b_H, 1, n, n)
+        self.M0rev = cache(lambda Hnh: self.I + 1j * self.dt * Hnh)
+
+        if self.options.normalize:
+            # `R` is close to identity but not exactly, we inverse it to normalize the
+            # Kraus map in order to have a trace-preserving scheme
+            self.R = cache(lambda M0: M0.mH @ M0 + self.dt * self.sum_LdagL)
+            self.Rrev = cache(lambda M0rev: M0rev.mH @ M0rev - self.dt * self.sum_LdagL)
+
+            if self.H.is_constant:
+                # if H is constant, we normalize the operators once by computing
+                # `S = sqrt(R)^-1` and applying is to the `Ms` operators
+                H = self.H(0.0)
+                Hnh = self.Hnh(H)
+                M0 = self.M0(Hnh)
+                M0rev = self.M0rev(Hnh)
+
+                S = inv_sqrtm(self.R(M0))
+                Srev = inv_sqrtm(self.Rrev(M0rev))
+
+                self.M0 = cache(lambda Hnh: M0 @ S)
+                self.M1s = self.M1s @ S
+                self.M0rev = cache(lambda Hnh: M0rev @ Srev)
```

Comment:
Also, it should be
```python
self.M0rev = cache(lambda Hnh: Srev @ M0rev)
self.M1srev = Srev @ self.M1s
```
I think in order to preserve the unit Kraus map structure.